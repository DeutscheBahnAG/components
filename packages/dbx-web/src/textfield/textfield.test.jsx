import React from 'react';
import { shallow } from 'enzyme';
import Textfield from './textfield';

describe('Textfield Component', () => {
  it('should render a text input', () => {
    const wrapper = shallow(<Textfield value="Text" />);
    expect(wrapper.find('input[type="text"]')).toHaveLength(1);
  });

  it('should render a number input', () => {
    const wrapper = shallow(<Textfield type="number" value="Text" />);
    expect(wrapper.find('input[type="number"]')).toHaveLength(1);
  });

  it('should render a textarea', () => {
    const wrapper = shallow(<Textfield type="textarea" value="Text" />);
    expect(wrapper.find('textarea')).toHaveLength(1);
  });

  it('should render a disabled textarea', () => {
    const wrapper = shallow(<Textfield type="textarea" disabled value="Text" />);
    expect(wrapper.find('.dbx-textfield.dbx-textfield--disabled')).toHaveLength(1);
    expect(wrapper.find('textarea[disabled]')).toHaveLength(1);
  });

  it('should render a disabled input', () => {
    const wrapper = shallow(<Textfield disabled value="Text" />);
    expect(wrapper.find('.dbx-textfield.dbx-textfield--disabled')).toHaveLength(1);
    expect(wrapper.find('input[disabled]')).toHaveLength(1);
  });

  it('should render a read-only input', () => {
    const wrapper = shallow(<Textfield readOnly value="Text" />);
    expect(wrapper.find('.dbx-textfield.dbx-textfield--readonly')).toHaveLength(1);
    expect(wrapper.find('input[readOnly]')).toHaveLength(1);
  });

  it('should use a prefix', () => {
    const wrapper = shallow(<Textfield prefix="$" value="Text" />);
    expect(wrapper.find('.dbx-textfield-prefix')).toHaveLength(1);
  });

  it('should use a suffix', () => {
    const wrapper = shallow(<Textfield suffix="€" value="Text" />);
    expect(wrapper.find('.dbx-textfield-suffix')).toHaveLength(1);
  });

  it('should use an unit before the field', () => {
    const wrapper = shallow(<Textfield unit="£" value="Text" />);
    expect(wrapper.find('.dbx-textfield-prefix')).toHaveLength(1);
  });

  it('should use an unit after the field', () => {
    const wrapper = shallow(<Textfield unit="€" value="Text" />);
    expect(wrapper.find('.dbx-textfield-suffix')).toHaveLength(1);
  });
});
