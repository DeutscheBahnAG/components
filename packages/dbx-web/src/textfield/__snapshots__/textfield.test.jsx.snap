// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield Component should render a disabled input 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled dbx-textfield--disabled"
  >
    <input
      disabled=""
      type="text"
      value="My disabled 2"
    />
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should render a disabled textarea 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--textarea dbx-textfield--xl dbx-textfield--filled dbx-textfield--disabled"
  >
    <textarea
      disabled=""
    >
      My disabled 1
    </textarea>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should render a number input 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--number dbx-textfield--xl dbx-textfield--filled"
  >
    <input
      type="number"
      value="123"
    />
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should render a read-only input 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled dbx-textfield--readonly"
  >
    <input
      readonly=""
      type="text"
      value="Read Only"
    />
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should render a text input 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled"
  >
    <input
      type="text"
      value="My value 1"
    />
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should render a textarea 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--textarea dbx-textfield--xl dbx-textfield--filled"
  >
    <textarea>
      Textarea
    </textarea>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should use a prefix 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled"
  >
    <span
      class="dbx-textfield-prefix"
    >
      $
    </span>
    <input
      type="text"
      value="Text"
    />
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should use a suffix 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled"
  >
    <input
      type="text"
      value="Text"
    />
    <span
      class="dbx-textfield-suffix"
    >
      €
    </span>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should use an unit after the field 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled"
  >
    <input
      type="text"
      value="Text"
    />
    <span
      class="dbx-textfield-suffix"
    >
      €
    </span>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Textfield Component should use an unit before the field 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--text dbx-textfield--xl dbx-textfield--filled"
  >
    <span
      class="dbx-textfield-prefix"
    >
      £
    </span>
    <input
      type="text"
      value="Text"
    />
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;
